beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -328,6 +330,48 @@ pub enum BlockError<E: EthSpec> {
     InternalError(String),
 }
 
+impl<E: EthSpec> BlockError<E> {
+    pub fn rpc_scoring(&self) -> ErrorCategory {
+        match self {
+            Self::ParentUnknown { .. } => ErrorCategory::internal_recoverable(),
+            Self::FutureSlot { .. } => ErrorCategory::internal_recoverable(),
+            Self::StateRootMismatch { .. } => ErrorCategory::malicious_non_recoverable(),
+            Self::GenesisBlock => ErrorCategory::internal_non_recoverable(),
+            Self::WouldRevertFinalizedSlot { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::NotFinalizedDescendant { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::BlockIsAlreadyKnown { .. } => ErrorCategory::internal_non_recoverable(),
+            Self::BlockSlotLimitReached => ErrorCategory::malicious_non_recoverable(),
+            Self::IncorrectBlockProposer { .. } => ErrorCategory::malicious_non_recoverable(),
+            Self::ProposalSignatureInvalid => ErrorCategory::malicious_recoverable(),
+            Self::UnknownValidator { .. } => ErrorCategory::malicious_non_recoverable(),
+            // TODO: Labeling as recoverable as it may be the proposer signature. We should check
+            // Which one is it and label as non recoverable if the proposer signature is correct.
+            Self::InvalidSignature => ErrorCategory::malicious_recoverable(),
+            Self::BlockIsNotLaterThanParent { .. } => ErrorCategory::malicious_non_recoverable(),
+            // TODO: Not clear what to set here
+            Self::NonLinearParentRoots => ErrorCategory::malicious_recoverable(),
```

Comment:
this would be malicious non recoverable since we are checking the block root matches what we requested